common/paramsbuilder/params.go

```diff
@@ -0,0 +1,104 @@
+package paramsbuilder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+	"golang.org/x/oauth2"
+)
+
+// ParamAssurance checks that param data is valid
+// Every param instance must implement it.
+type ParamAssurance interface {
+	ValidateParams() error
```

Comment:
There are several parameters with single responsibilities: Client/Workspace/Module. There could be more of these shared parameters.
Every parameter must determine if its value is required/optional by implementing `ParamAssurance`.

Reply:
This is a very cool package, thanks for introducing it!